src/BizHawk.Common/DeltaSerializer.cs

```diff
@@ -20,8 +20,19 @@ namespace BizHawk.Common
 	/// </summary>
 	public static class DeltaSerializer
 	{
+		public static int GetMaxDeltaSize<T>(int count)
+			where T : unmanaged =>
+			Marshal.SizeOf<T>() * count;

```

Comment:
sizeof() should be used, not Marshal.SizeOf<> (which has different semantics and is much slower)